### PR TITLE
update config module of stable_diffusion_v2

### PR DESCRIPTION
**Pull request by The-truthh**

### CODE REVIEW

File: examples/stable_diffusion_v2/ldm/modules/train/optim.py [hunk 1]
```diff
@@ -1,28 +1,16 @@
...
-# limitations under the License.
-# ============================================================================
 """
  build optimizer for ms
 """
 from mindspore.nn.optim.adam import Adam, AdamWeightDecay
 
 
-def build_optimizer(model, opts, lr):
+def build_optimizer(model, optim, betas, weight_decay, lr):
```
Comment (DongyaoZhu) [inline examples/stable_diffusion_v2/ldm/modules/train/optim.py:7]: not all optim use betas? I'd go for **optim_kwargs

File: examples/stable_diffusion_v2/train_dreambooth.py [hunk 2]
```diff
@@ -90,16 +103,10 @@ def parse_args():
...
     )
     parser.add_argument(
-        "--train_config",
-        default="configs/train_config.json",
-        type=str,
-        help="Specify the path to the train config file",
-    )
-    parser.add_argument(
-        "--pretrained_model_path", default="", type=str, help="Specify the directory of the pretrained model"
+        "--model_config", default="configs/train_dreambooth_sd_v2.yaml", type=str, help="model config path"
```
Comment (DongyaoZhu) [inline examples/stable_diffusion_v2/train_dreambooth.py:106]: is there a reason we're not using config/train/*yamls here?

**Review by DongyaoZhu — COMMENTED** [review]
see comments

File: examples/stable_diffusion_v2/ldm/modules/train/optim.py (hunk 1 above)
Comment (The-truthh) [reply examples/stable_diffusion_v2/ldm/modules/train/optim.py:7]: Both optimizers implemented require parameter betas, and I think all the parameters should be used explicitly in train_*.py

File: examples/stable_diffusion_v2/train_dreambooth.py (hunk 2 above)
Comment (The-truthh) [reply examples/stable_diffusion_v2/train_dreambooth.py:106]: We still use config/train/*yamls by train_config argument, which is the first argument in the parser

File: examples/stable_diffusion_v2/README.md [hunk 3]
```diff
@@ -324,37 +326,45 @@ It is simple to switch from SD 2.0 to SD 1.5 by setting the `--version` (`-v`) a
...
 
 ```shell
 # SD 1.5 vanilla training
-sh scripts/run_train_v1.sh
+python train_text_to_image.py \
+    --train_config "configs/train/train_config_vanilla_v1_chinese.yaml" \
+    --data_path "datasets/pokemon_cn/train" \
+    --output_path "output/txt2img" \
+    --pretrained_model_path "models/wukong-huahua-ms.ckpt"
 ```
```
Comment (SamitHuang) [inline examples/stable_diffusion_v2/README.md:360]: Please add instructions on running SD1.5 en training as well

File: examples/stable_diffusion_v2/train_text_to_image.py [hunk 4]
```diff
@@ -140,22 +127,164 @@ def load_pretrained_model_vae_unet_cnclip(pretrained_ckpt, cnclip_ckpt, net):
...
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--train_config",
+        default="",
+        type=str,
+        help="train config path to load a yaml file that override the default arguments",
+    )
+    parser.add_argument(
+        "-v",
+        "--version",
```
Comment (SamitHuang) [inline examples/stable_diffusion_v2/train_text_to_image.py:148]: How is --version affect the pretrained model weight selection here? I think we should either delete it or use it to select the pretrained weight version to finetune from

**Review by SamitHuang — APPROVED** [review]

**Review by geniuspatrick — APPROVED** [review]

**Pull request merged**